front/app/containers/IdeasShow/index.tsx

```diff
@@ -197,20 +197,17 @@ export const IdeasShow = ({
   const [translateButtonIsClicked, setTranslateButtonIsClicked] =
     useState<boolean>(false);
   const [queryParams] = useSearchParams();
+  const ideaIdParameter = queryParams.get('new_idea_id');
+  const timeout = useRef<NodeJS.Timeout>();
 
   useEffect(() => {
-    const newIdeaId = queryParams.get('new_idea_id');
-    let timeout: NodeJS.Timeout;
-    if (isString(newIdeaId)) {
-      timeout = setTimeout(() => {
-        setNewIdeaId(newIdeaId);
+    if (isString(ideaIdParameter)) {
+      timeout.current = setTimeout(() => {
+        setNewIdeaId(ideaIdParameter);
       }, 1500);
       clHistory.replace(window.location.pathname);
     }
-    return () => {
```

Comment:
It doesn't work to clear it here, but when I clear it later in the close modal function instead it does.

Reply:
I'm missing the context here a bit... the `clHistory.replace` call is there to remove the parameters from the URL, right? Is there any harm in leaving them in there? And why do you only replace them after a timeout?

Reply:
I discussed that with Ben a while back, and it's mostly a UX thing - we don't really want long URL parameters shown to the resident. The timeout is indeed annoying - it's required to allow for the modal animation to happen (where it moves down from the top). Ideally we update the modal component itself to handle the animation, but we haven't had time for that. In other places in the platform this is also handled using a timeout :) I think the issue in this case was that we were clearing the timeout too soon, but if we clear it when we close the modal instead it works properly.